packages/design-system/src/components/client/toggle.tsx

```diff
@@ -0,0 +1,14 @@
+import { Switch as SwitchHeadless, type SwitchProps as SwitchPropsHeadless } from "@headlessui/react";
+import * as React from "react";
+
+// biome-ignore lint/suspicious/noExplicitAny: allow any for generic `as` prop support
+export type Toggle = SwitchPropsHeadless<any>;
+
+// biome-ignore lint/suspicious/noExplicitAny: allow any for loose ref signature
+function ToggleComponent(props: Toggle, ref: React.Ref<any>) {
+  return <SwitchHeadless {...props} ref={ref} />;
+}
+
+const Toggle = React.forwardRef(ToggleComponent) as unknown as typeof SwitchHeadless;
```

Comment:
Kdyz fixnes any, tak to nebude potreba.

Pokud by tam precijen bylo z nejakeho duvodu vic moznosti, tak bych to stejne radsi natypoval. At je tam presne otypovany vyber tech komponent (ten Toggle tam ma v generiku jakykoli ElementRef).